### PR TITLE
Remove old mailers

### PR DESCRIPTION
**Pull request by megcrenshaw**
Removes the Registration Sent and Waitlist teacher application mailers as per https://codedotorg.atlassian.net/browse/ACQ-178.

When doing this, I also removed all other instances of `registration_sent` in our repo, and the cohort view now only includes `accepted` status, which is Requirement 8 of https://docs.google.com/document/d/1HedK5_qh8-Yl6bFwfyU0ZB-UHgQK7l8-4trJhoGYI-8/edit#.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/code-studio/pd/application_dashboard/summaryTest.js [hunk 1]
```diff
@@ -10,16 +10,14 @@ import sinon from 'sinon';
...
-    waitlisted: {locked: 0, total: 0},
+    pending_space_availability: {locked: 0, total: 0},
+    accepted: {locked: 0, total: 0},
     declined: {locked: 0, total: 0},
-    accepted_not_notified: {locked: 0, total: 0},
-    accepted_notified_by_partner: {locked: 0, total: 0},
-    accepted_no_cost_registration: {locked: 0, total: 0},
-    registration_sent: {locked: 0, total: 0},
-    paid: {locked: 0, total: 0}
+    withdrawn: {locked: 0, total: 0}
```
Comment (megcrenshaw) [inline apps/test/unit/code-studio/pd/application_dashboard/summaryTest.js:20]: Made these consistent with our summary view, removing statuses that we don't have

File: lib/cdo/shared_constants/pd/shared_application_constants.rb [hunk 2]
```diff
@@ -22,13 +22,6 @@ module SharedApplicationConstants
...
     COHORT_CALCULATOR_STATUSES = %w(
       accepted
-      accepted_not_notified
-      accepted_notified_by_partner
-      accepted_no_cost_registration
-      registration_sent
-      paid
     )
-
-    COHORT_VIEW_STATUSES = COHORT_CALCULATOR_STATUSES + ['withdrawn']
```
Comment (megcrenshaw) [inline lib/cdo/shared_constants/pd/shared_application_constants.rb:32]: This is part of Requirement 8 in https://docs.google.com/document/d/1HedK5_qh8-Yl6bFwfyU0ZB-UHgQK7l8-4trJhoGYI-8/edit#

File: dashboard/lib/services/registration_reminder.rb [hunk 3]
```diff
@@ -22,22 +22,22 @@ def self.queue_registration_reminders!
...
   # @return [Enumerable<Pd::Application::ApplicationBase>]
   def self.applications_needing_first_reminder
-    # The 'registration_sent' email was sent at least two weeks ago
+    # The 'accepted_no_cost_registration' email was sent at least two weeks ago
     # No 'registration_reminder' has been sent yet.
-    # Not enrolled in a workshop since the 'registration_sent' email was sent
+    # Not enrolled in a workshop since the 'accepted_no_cost_registration' email was sent
     applications_awaiting_enrollment.
-      where("registration_sent.sent_at <= ?", 2.weeks.ago).
+      where("accepted_no_cost_registration.sent_at <= ?", 2.weeks.ago).
```
Comment (megcrenshaw) [inline dashboard/lib/services/registration_reminder.rb:29]: @TurnerRiley I believe you're going to be updating the mailers for `accepted` statuses –– not sure if there will be reminder emails if they haven't enrolled, but this is the code that makes that happen.

Comment (TurnerRiley) [reply dashboard/lib/services/registration_reminder.rb:29]: Gotcha, thank you!

**Review by TurnerRiley — APPROVED** [review]
Looks fantastic! Thanks for tracking all the moving parts with this!

**Pull request merged**